test/support/shared_examples/on_duplicate_key_update.rb

```diff
@@ -284,6 +284,27 @@ def should_support_basic_on_duplicate_key_update
         end
       end
 
+      context "with an association that has a non-standard primary key" do
+        let(:columns) { [:id, :token] }
+        let(:values) { [[5, '12345abcedf67890']] }
+        let(:update_columns) { [:token] }
+
+        macro(:perform_import) do |*opts|
+          @user.user_tokens.import columns, values, opts.extract_options!.merge(on_duplicate_key_update: update_columns, validate: false)
+        end
+        macro(:created_token) { UserToken.find(5) }
+
+        setup do
+          User.import [User.new(id: 1, name: 'Solomon')], validate: false
+          @user = User.find 1
```

Comment:
This could be simplified a bit:

` @user = User.create(id: 1, name: 'Solomon')`